OpenRA.Mods.Common/Util.cs

```diff
@@ -175,5 +175,23 @@ public static int RandomDelay(World world, int[] range)
 
 			return world.SharedRandom.Next(range[0], range[1]);
 		}
+
+		static WDist allMinimumRequiredScanRadius;
+		static WDist blockerMinimumRequiredScanRadius;
+		static Ruleset lastRules = null;
+
+		public static WDist MinimumRequiredScanRadius(Ruleset rules, ActorInfo ai, bool onlyBlockers = false)
+		{
+			if (lastRules != rules)
```

Comment:
I suppose I am more concerned with accumulation (ms here, ms there) and we could always evaluate it later. There is also the question of if this should be in `ActorMapInfo` since actor-map information is its purpose. A trait lookup would also have an epsilon cost, but the the function would be in a more meaningful place.